File: src/main/java/wzq/jcstress/plugin/configuration/JCStressConfigurationType.java
package wzq.jcstress.plugin.configuration;

import com.intellij.execution.configurations.JavaRunConfigurationModule;
import com.intellij.execution.configurations.RunConfiguration;
import com.intellij.execution.configurations.SimpleConfigurationType;
import com.intellij.icons.AllIcons;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.util.NotNullLazyValue;
import org.jetbrains.annotations.NotNull;

/**
 * @author 吴志强
 * @date 2022/8/24
 */
public class JCStressConfigurationType extends SimpleConfigurationType {

    public JCStressConfigurationType() {
        super("jcstress-id", "JCStress", "Configuration to run a jcstress test", NotNullLazyValue.createValue(() -> AllIcons.Actions.ProfileYellow));
    }

    @Override
    public @NotNull RunConfiguration createTemplateConfiguration(@NotNull Project project) {
        return new JCStressRunConfiguration("jcstress", new JavaRunConfigurationModule(project, false), this);
    }
}

File: src/main/kotlin/wzq/jcstress/plugin/JCStressConfigurationType.kt
package wzq.jcstress.plugin

import com.intellij.execution.configurations.JavaRunConfigurationModule
import com.intellij.execution.configurations.RunConfiguration
import com.intellij.execution.configurations.SimpleConfigurationType
import com.intellij.openapi.project.Project
import com.intellij.openapi.util.NotNullFactory
import com.intellij.openapi.util.NotNullLazyValue
import wzq.jcstress.plugin.configuration.Icons

/**
 * @author 吴志强
 */
class JCStressConfigurationType :
    SimpleConfigurationType(
        "jcstress",
        "JCStress",
        "Configuration to run a jcstress test",
        NotNullLazyValue.createValue(NotNullFactory { Icons.CONFIG }),
    ) {
    override fun createTemplateConfiguration(project: Project): RunConfiguration =
        JCStressRunConfiguration("jcstress", JavaRunConfigurationModule(project, false), this)
}
